src/components/MapStyleInput.svelte

```diff
@@ -1,62 +1,155 @@
 <script>
+  import hat from 'hat';
   import { onDestroy, onMount } from 'svelte';
   import {
     maps as mapsStore,
     stylePresets as stylePresetsStore,
     config as configStore,
   } from '../stores';
-  import { createBranchUrl, parseBranchUrl } from '../branch-utils';
+  import { createBranchUrl } from '../branch-utils';
   import { shortcut } from '../shortcut';
   import { fetchUrl } from '../fetch-url';
 
   export let index;
 
   let map;
   let url;
-  let name;
   let branch;
-  let pattern;
-  let allowPolling = true;
 
-  let stylePresets;
   let branchPatterns;
-
-  stylePresetsStore.subscribe(value => (stylePresets = value));
   configStore.subscribe(value => ({ branchPatterns } = value));
+  mapsStore.subscribe(maps => {
+    map = maps.find(m => m.index === index);
+    if (map) {
+      branch = map.branch;
+      url = map.url;
+    }
+  });
 
   let selected;
   let textInput = url;
-  let dropdownOptions = {};
 
-  onMount(() => {
-    // This runs only on mount to check for localhost in url
-    poll(url);
-  });
+  let focused = false;
+  let error = null;
 
+  // Used to create dropdown display with headings
+  let dropdownDisplayOptions = {};
+  // Actual values selected from dropdown matched by id
+  let dropdownValues = [];
```

Comment:
I'm open to reverting this if it actually feels _less_ clear, but I like having something more standardized to pass into our select element below, and I feel like this may open up opportunity for standardizing the values more as well across presets, branch, and custom.